tests/test_generator.py

```diff
@@ -42,32 +53,37 @@ def test_process_class(
             [mock.call(type_o), mock.call(type_m), mock.call(type_n)]
         )
 
-        mock_process_attribute.assert_has_calls(
-            [
-                mock.call(e.attrs[0], ["@a", "@e"]),
-                mock.call(e.attrs[1], ["@a", "@e"]),
-                mock.call(e.attrs[2], ["@a", "@e"]),
-                mock.call(i.attrs[0], ["@a", "@i"]),
-                mock.call(i.attrs[1], ["@a", "@i"]),
-                mock.call(i.attrs[2], ["@a", "@i"]),
-                mock.call(a.attrs[0], ["@a"]),
-                mock.call(a.attrs[1], ["@a"]),
-                mock.call(a.attrs[2], ["@a"]),
-            ]
+        mock_process_attributes.assert_has_calls(
+            [mock.call(i, ["@a", "@i"]), mock.call(a, ["@a"])]
         )
-        self.assertEqual(9, mock_process_attribute.call_count)
+        mock_process_enumerations.assert_called_once_with(e)
 
     @mock.patch.object(generator, "process_enumeration")
-    def test_process_enum_class(self, mock_process_enumeration):
-        a = ClassFactory.create(
-            name="a", attrs=AttrFactory.list(2, local_type=TagType.ENUMERATION)
+    def test_process_enumerations(self, mock_process_enumeration):
+        obj = ClassFactory.create(
+            attrs=[
+                AttrFactory.create(default=3),
+                AttrFactory.create(default=2),
+                AttrFactory.create(default=1),
+                AttrFactory.create(default=1),
+            ]
         )
-        generator.process_class(a)
+        calls = reversed([mock.call(attr, obj) for attr in obj.attrs])
 
-        mock_process_enumeration.assert_has_calls(
-            [mock.call(a.attrs[0], a), mock.call(a.attrs[1], a)]
+        generator.process_enumerations(obj)
+        mock_process_enumeration.assert_has_calls(calls)
+        self.assertEqual(1, obj.attrs[0].default)
+        self.assertEqual(2, obj.attrs[1].default)
+        self.assertEqual(3, obj.attrs[2].default)
+        self.assertEqual(3, len(obj.attrs))
+
+    @mock.patch.object(generator, "process_attribute")
+    def test_process_attributes(self, mock_process_attribute):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/tefra/xsdata/pullRequest?prid=5052528)